tests/Dfe.PlanTech.AzureFunctions.UnitTests/AsyncEnumerator.cs

```diff
@@ -23,6 +23,7 @@ public ValueTask<bool> MoveNextAsync()
 
     public ValueTask DisposeAsync()
     {
+        GC.SuppressFinalize(this);
```

Comment:
what does this do?

Reply:
It just tells the garbage collector to not bother putting the object into the finaliser queue, as we've already cleaned everything up in our disposing method. It's part of the standard C# `IDisposable` pattern implementation.

That being said:
1. There's nothing to dispose of here, because it's not a "real" implementation of anything.
2. I haven't properly implemented the `IDisposable` pattern still, but SonarCloud wasn't complaining about the rest of it.

I debated implementing the full pattern "properly", but since there's nothing to actually dispose of it felt unnecessary.

FWIW the full pattern would end up being something like:

```csharp
namespace Dfe.PlanTech.AzureFunctions.UnitTests;

public class AsyncEnumerator<T> : IAsyncEnumerator<T>
{
    private bool disposed;
    private readonly IEnumerator<T> _inner;

    public AsyncEnumerator(IEnumerator<T> inner)
    {
        _inner = inner;
    }

    public void Dispose()
    {
        _inner.Dispose();
    }

    public T Current => _inner.Current;

    public ValueTask<bool> MoveNextAsync()
    {
        return ValueTask.FromResult(_inner.MoveNext());
    }

    public ValueTask DisposeAsync()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
        return ValueTask.CompletedTask;
    }

    protected virtual void Dispose(bool disposing)
    {
        if (!disposed)
        {
            if (disposing)
            {
                _inner.Dispose();
            }

            disposed = true;
        }
    }

    ~AsyncEnumerator()
    {
        Dispose(false);
    }
}
```